app/_src/kubernetes-ingress-controller/guides/allowing-multiple-authentication-methods.md

```diff
@@ -9,414 +9,326 @@ for all requests without regard for whether a request has been authenticated
+    config:
+      anonymous: anonymous
+      hide_credentials: true
+    plugin: basic-auth
+    ' | kubectl apply -f -
+    ```
+    The results should look like this: 
+    ```text
+    kongplugin.configuration.konghq.com/httpbin-basic-auth created
+    ```
+    Create a plugin named `httpbin-kay-auth`.
+    ```bash
+    echo '
+    apiVersion: configuration.konghq.com/v1
+    kind: KongPlugin
+    metadata:
+      name: httpbin-key-auth
+    config:
+      key_names:
+        - apikey
+      anonymous: anonymous
+      hide_credentials: true
+    plugin: key-auth
+    ' | kubectl apply -f -
+    ```
+    The results should look like this:
+    ```text
+    kongplugin.configuration.konghq.com/httpbin-key-auth created
+    ```
+
+1. Associate the plugins with the Ingress rule that you created.
+
+    ```bash
+    echo '
+    apiVersion: networking.k8s.io/v1
+    kind: Ingress
+    metadata:
+      name: demo
+      annotations:
+        konghq.com/strip-path: "true"
+        konghq.com/plugins: httpbin-basic-auth, httpbin-key-auth
+    spec:
+      ingressClassName: kong
+      rules:
+      - http:
+          paths:
+          - path: /test
+            pathType: Prefix
+            backend:
+              service:
+                name: httpbin
+                port:
+                  number: 80
+    ' | kubectl apply -f -
+    ```
+    The results should look like this:
+    ```text
+    ingress.networking.k8s.io/demo configured
+    ```
+
+    At this point unauthenticated requests and requests with invalid credentials are still allowed. The anonymous consumer is allowed, and is applied to any request that does not pass a set of credentials associated with some other consumer.
+
+1. Test unauthenticated requests.
+
+    ```bash
+    curl -i $PROXY_IP/test/status/200
+    ```
+    The results should look like this:
+    ```text
+    HTTP/1.1 200 OK
+    Content-Type: text/html; charset=utf-8
+    Content-Length: 0
+    Connection: keep-alive
+    WWW-Authenticate: Key realm="kong"
+    Server: gunicorn/19.9.0
+    Access-Control-Allow-Origin: *
+    Access-Control-Allow-Credentials: true
+    X-Kong-Upstream-Latency: 2488
+    X-Kong-Proxy-Latency: 4
+    Via: kong/3.3.1
+    ```
+1. Test invalid credentials requests.
+    ```bash
+    curl -i $PROXY_IP/test/status/200 -H "apikey=invalid"
+    ```
+   The results should look like this:
+    ```text
+    HTTP/1.1 200 OK
+    Content-Type: text/html; charset=utf-8
+    Content-Length: 0
+    Connection: keep-alive
+    WWW-Authenticate: Key realm="kong"
+    Server: gunicorn/19.9.0
+    Access-Control-Allow-Origin: *
+    Access-Control-Allow-Credentials: true
+    X-Kong-Upstream-Latency: 1031
+    X-Kong-Proxy-Latency: 1
+    Via: kong/3.3.1
+    ```
+
+1. To add a `key-auth` credential for one consumer, and a `basic-auth`credential for another you need to create a [Secret](https://kubernetes.io/docs/concepts/configuration/secret/) resource with an API-key, username, and password.
```

Comment:
Why do we configure `consumer-2` before `consumer-1`?

Reply:
Yes!!! as i was writing I did think on that. Will reverse the order.